bloomstack_core/public/js/bloomstack_core/admin_insights/Home.vue

```diff
@@ -252,164 +209,99 @@
+      TabItemProductCount: QUERY.TabItemProductCount,
+      TabSalesInvoiceRevnue: QUERY.TabSalesInvoiceRevnue,
+      TabSalesInvoiceTopCustomerByRevenue:
+        QUERY.TabSalesInvoiceTopCustomerByRevenue,
+      TabSalesInvoiceTopCustomerGroupByRevenue:
+        QUERY.TabSalesInvoiceTopCustomerGroupByRevenue,
+      TabSalesInvoiceTopsalesPartnerByRevenue:
+        QUERY.TabSalesInvoiceTopsalesPartnerByRevenue,
+      TabSalesInvoiceRevenueByTerritory:
+        QUERY.TabSalesInvoiceRevenueByTerritory,
     };
-    return { ...dataObj, startDate, endDate, selectedDateRange };
-  },
-  watch: {
-    startDate () {
-      this.selectedDateRange = { value: 0, text: null };
-      this.dateRange();
-    },
-    endDate () {
-      this.selectedDateRange = { value: 0, text: null };
-      this.dateRange();
-    },
-    methods: {
-      dateRange () {
-        return {
-          startDate: this.startDate,
-          endDate: this.endDate,
-        };
-      },
-    },
-  },
-  computed: {
-    tabSalesInvoiceRevnue() {
-      return Query.tabSalesInvoiceRevnue(
-        this.startDate,
-        this.endDate,
-        this.selectedDateRange.text
-      );
-    },
-
-    tabSalesInvoiceTopCustomerByRevenue() {
-      return Query.tabSalesInvoiceTopCustomerByRevenue(
-        this.startDate,
-        this.endDate,
-        this.selectedDateRange.text
-      );
-    },
-
-    tabSalesInvoiceTopCustomerGroupByRevenue() {
-      return Query.tabSalesInvoiceTopCustomerGroupByRevenue(
-        this.startDate,
-        this.endDate,
-        this.selectedDateRange.text
-      );
-    },
-
-    tabSalesInvoiceTopsalesPartnerByRevenue() {
-      return Query.tabSalesInvoiceTopsalesPartnerByRevenue(
-        this.startDate,
-        this.endDate,
-        this.selectedDateRange.text
-      );
-    },
-
-    tabSalesInvoiceRevenueByTerritory() {
-      return Query.tabSalesInvoiceRevenueByTerritory(
-        this.startDate,
-        this.endDate,
-        this.selectedDateRange.text
-
-      );
-    },
-
-    tabSalesInvoiceWeeklySales() {
-      return Query.tabSalesInvoiceWeeklySales(this.startDate, this.endDate,this.selectedDateRange.text);
-    },
-    tabItemProductCount() {
-      return Query.tabItemProductCount;
-    },
-
-    tabSalesInvoiceItemTrueQty() {
-      return Query.tabSalesInvoiceItemTrueQty;
-    },
-
-    tabSalesInvoiceTrueCount() {
-      return Query.tabSalesInvoiceTrueCount;
-    },
-
-    tabPaymentEntryWeeklyAverage() {
-      return Query.tabPaymentEntryWeeklyAverage;
-    },
-
-    tabPaymentEntryMonthlyAverage(){
-      return Query.tabPaymentEntryMonthlyAverage;
-    },
-
-    tabSalesInvoiceItemConversionRatio(){
-      return Query.tabSalesInvoiceItemConversionRatio;
-    },
-
-    tabPaymentEntryWeeklyRevnue(){
-      return Query.tabPaymentEntryWeeklyRevnue;
-    },
-
-    tabPaymentEntryMonthlyRevnue(){
-      return Query.tabPaymentEntryMonthlyRevnue;
-    }
+    return { ...dataObj, startDate, endDate };
```

Comment:
if you do these changes then how we will paas startDate and endDate to a particular Query